discretisedfield/tests/test_region.py

```diff
@@ -278,10 +296,65 @@ def test_mul_truediv(self):
         with pytest.raises(TypeError):
             res = 5 / region
 
+    def test_units(self):
+        p1 = (-50e-9, -50e-9, 0)
+        p2 = (50e-9, 50e-9, 20e-9)
+        ndim = 3
+        units = ["m", "m", "m"]
```

Comment:
I would suggest to test some non-default values here (and also the default if not provided explicitly.